bin/run.js

```diff
@@ -228,39 +226,45 @@ if (coverageReport && !global.__coverage__ && files.length === 0) {
   var args = [nycBin, 'report', '--reporter', coverageReport]
   var child
 
-  // automatically hook into coveralls
+  // automatically hook into coveralls and/or codecov
   if (coverageReport === 'text-lcov' && pipeToService) {
     child = spawn(node, args)
-    var covBin, covName
-
-    if (process.env.COVERALLS_REPO_TOKEN) {
-      covBin = coverallsBin
-      covName = 'Coveralls'
-    } else if (process.env.CODECOV_TOKEN) {
-      covBin = codecovBin
-      covName = 'Codecov'
-    }
 
-    var ca = spawn(node, [covBin], {
-      stdio: [ 'pipe', 1, 2 ],
-      env: process.env
-    })
-    child.stdout.pipe(ca.stdin)
-    ca.on('close', function (code, signal) {
-      if (signal)
-        process.kill(process.pid, signal)
-      else if (code)
-        process.exit(code)
-      else
-        console.log('Successfully piped to ' + covName)
+    var services = [
+      process.env.COVERALLS_REPO_TOKEN && {
+        covBin: require.resolve('coveralls/bin/coveralls.js')
+      , covName: 'Coveralls'
+      }
+    , process.env.CODECOV_TOKEN && {
+        covBin: require.resolve('codecov.io/bin/codecov.io.js')
+      , covName: 'Codecov'
+      }
+    ].filter(function(s) {
+      return !!s // remove undefined services
     })
-    signalExit(function (code, signal) {
-      child.kill('SIGHUP')
-      ca.kill('SIGHUP')
+
+    services.forEach(function(s) {
+      var ca = spawn(node, [s.covBin], {
+        stdio: [ 'pipe', 1, 2 ],
+        env: process.env
+      })
+      child.stdout.pipe(ca.stdin)
+      ca.on('close', function (code, signal) {
+        if (signal)
+          process.kill(process.pid, signal)
+        else if (code)
+          process.exit(code)
+        else
+          console.log('Successfully piped to ' + s.covName)
+      })
+      signalExit(function (code, signal) {
+        child.kill('SIGHUP')
+        ca.kill('SIGHUP')
+      })
     })
   } else {
     // otherwise just run the reporter
-    var child = fg(node, args)
+    child = fg(node, args)
```

Comment:
why `var` when we do it on [line 227](https://github.com/isaacs/node-tap/pull/183/files#diff-1316baf4c35bbaca22d1c8c3ab02bb59R227)